### ISSUE TITLE
Fix failing tests

### ISSUE THREAD

**Issue by timlinux** [kartoza/stream_feature_extractor#35]
# Problem

We have 2 failing tests
# Details

```
timlinux@confluence:~/dev/python/stream-feature-extractor$ make test

----------------------------------------
Compiled translation files to .qm files.
----------------------------------------
safe_qgis/i18n/inasafe_en.ts
Updating 'i18n/en.qm'...
    Generated 0 translation(s) (0 finished and 0 unfinished)
    Ignored 40 untranslated source text(s)
safe_qgis/i18n/inasafe_af.ts
Updating 'i18n/af.qm'...
    Generated 1 translation(s) (1 finished and 0 unfinished)
    Ignored 39 untranslated source text(s)
safe_qgis/i18n/inasafe_de.ts
Updating 'i18n/de.qm'...
    Generated 40 translation(s) (15 finished and 25 unfinished)
safe_qgis/i18n/inasafe_id.ts
Updating 'i18n/id.qm'...
    Generated 8 translation(s) (8 finished and 0 unfinished)
    Ignored 32 untranslated source text(s)

----------------------
Regression Test Suite
----------------------
export PYTHONPATH=`pwd`:`pwd`/third_party:/usr/local/qgis-2.0/share/qgis/python:/usr/local/qgis-2.0/share/qgis/python/plugins:; \
        export QGIS_DEBUG=0; \
        export QGIS_LOG_FILE=/dev/null; \
        nosetests -v --exclude pydev --with-id --with-coverage \
        --cover-package= . \
        3>&1 1>&2 2>&3 3>&- || true
#1 Test that the plugin __init__ will validate on plugins.qgis.org. ... ok
#2 Test that QGIS properly parses a wkt string. ... Loading a style file that was saved with an older version of qgis (saved in 1.9.0-Master, loaded in 2.0.1-Dufour). Problems may occur.
ok
#3 QGIS environment has the expected providers ... ok
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
#36 test for add_associated_nodes ... ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ok
#37 Test for check_associated_attributes ... ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ok
#38 Test for checking the data test. ... ok
#39 Test for creating nodes layer. ... ok
#40 Test for extracting nodes. ... ok
#41 Test for extracting nodes using dgn dataset. ... /usr/lib/python2.7/unittest/case.py:342: RuntimeWarning: TestResult has no addExpectedFailure method, reporting as passes
  RuntimeWarning)
ok
#42 Test for get_nearby_nodes function. ... ok
#43 Test for identify_branch method. ... ok
#44 Test for identify_confluence method. ... ok
#45 Test for identify_features on the dgn test dataset. ... ok
#46 Test for identify_features on the sungai baru dataset. ... FAIL
#47 Test for identify_pseudo_node method. ... ok
#48 Test for identify_segment_center. ... ok
#49 Test for identify_self_intersections. ... ok
#50 Test for identify_sink method. ... ok
#51 Test for identify_watershed method. ... ok
#52 Test for identify_well method. ... ok
#53 Test identify_intersections. ... FAIL
#54 test add_layer_attribute. ... ok
#55 test for list to str. ... ok
#56 test for get_random_string function. ... ok
#57 test for str_to_list. ... ok
#4 Test that translations work. ... ok

======================================================================
FAIL: Test for identify_features on the sungai baru dataset.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/timlinux/dev/python/stream-feature-extractor/test/test_stream_utilities.py", line 664, in test_identify_features_sungai_baru
    self.assertIn(result, expected_result, message)
AssertionError: [(4.50495e+06,5.81966e+06), u'Unseparated'] is not found 20
-------------------- >> begin captured stdout << ---------------------
Task progress: 20 of 20
Finding unclear bifurcations...
Task progress: 1 of 7
Finding branches...
Task progress: 2 of 7
Finding watersheds...
Task progress: 3 of 7
Finding confluences...
Task progress: 4 of 7
Finding sinks...
Task progress: 5 of 7
Finding pseudo nodes...
Task progress: 6 of 7
Finding wells...
Task progress: 7 of 7
Finding self intersections...
Task progress: 1 of 4
Finding segment centers...
Task progress: 2 of 4
Finding intersections...
Task progress: 3 of 4
Finding Unseparated...
Task progress: 4 of 4

--------------------- >> end captured stdout << ----------------------

======================================================================
FAIL: Test identify_intersections.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/timlinux/dev/python/stream-feature-extractor/test/test_stream_utilities.py", line 751, in test_intersections
    self.assertEqual(len(intersections), 22, '%s' % len(intersections))
AssertionError: 44

Coverage.py warning: Module  was never imported.
Name                                   Stmts   Miss  Cover   Missing
--------------------------------------------------------------------
ConfigParser                             391    391     0%   90-753
DLFCN                                     66     66     0%   2-83
OpenSSL                                   30     30     0%   4-44
OpenSSL.version                            2      2     0%   5-9
PyQt4                                      0      0   100%   
Queue                                    125    125     0%   3-244
StringIO                                 176    176     0%   31-324
UserDict                                 156    156     0%   3-180
__future__                                27     27     0%   50-126
_abcoll                                  378    378     0%   4-671
_sysconfigdata                             4      4     0%   1-6
_sysconfigdata_nd                          1      1     0%   2
_weakrefset                              152    152     0%   5-204
abc                                       82     82     0%   4-185
atexit                                    35     35     0%   8-65
base64                                   162    162     0%   3-360
bdb                                      421    421     0%   3-643
bisect                                    46     46     0%   3-92
calendar                                 385    385     0%   8-713
cgi                                      584    584     0%   12-1059
cmd                                      234    234     0%   48-404
codecs                                   397    397     0%   10-1095
collections                              290    290     0%   1-699
copy                                     299    299     0%   51-433
copy_reg                                 105    105     0%   7-188
coverage                                  33     33     0%   8-89
coverage.annotate                         57     57     0%   3-102
coverage.backward                         99     99     0%   9-184
coverage.bytecode                         42     42     0%   3-75
coverage.cmdline                         254    254     0%   3-744
coverage.codeunit                         70     70     0%   3-145
coverage.collector                       163    163     0%   3-353
coverage.config                           89     89     0%   3-213
coverage.control                         318    318     0%   3-779
coverage.data                            120    120     0%   3-278
coverage.debug                            27     27     0%   3-54
coverage.execfile                         88     88     0%   3-171
coverage.files                           159    159     0%   3-309
coverage.html                            194    194     0%   3-396
coverage.misc                             98     98     0%   3-167
coverage.parser                          337    337     0%   3-700
coverage.phystokens                      111    111     0%   3-210
coverage.report                           49     49     0%   3-92
coverage.results                         148    148     0%   3-286
coverage.summary                          64     64     0%   3-86
coverage.templite                        116    116     0%   5-208
coverage.version                           4      4     0%   4-9
coverage.xmlreport                        91     91     0%   3-155
ctypes                                   341    341     0%   4-555
ctypes._endian                            35     35     0%   4-64
ctypes.util                              189    189     0%   4-287
decimal                                 2485   2485     0%   20-6198
difflib                                  648    648     0%   29-2057
dis                                      177    177     0%   3-224
doctest                                  973    973     0%   9-2817
encodings                                 58     58     0%   31-157
encodings.aliases                          1      1     0%   18
encodings.ascii                           19     19     0%   9-42
encodings.hex_codec                       28     28     0%   9-71
encodings.unicode_escape                  16     16     0%   9-37
encodings.utf_8                           15     15     0%   9-34
fnmatch                                   65     65     0%   13-116
functools                                 43     43     0%   10-100
genericpath                               50     50     0%   6-105
gettext                                  379    379     0%   49-606
glob                                      53     53     0%   3-95
hashlib                                   58     58     0%   7-146
heapq                                    206    206     0%   3-485
hmac                                      54     54     0%   6-133
hotshot                                   34     34     0%   3-78
hotshot.log                              117    117     0%   1-194
hotshot.stats                             60     60     0%   3-96
httplib                                  704    704     0%   69-1342
inspect                                  574    574     0%   2-1059
io                                        23     23     0%   37-89
keyword                                   35     35     0%   3-93
linecache                                 80     80     0%   8-140
locale                                   371    371     0%   14-1947
logging                                  687    687     0%   17-1729
mimetools                                162    162     0%   4-250
multiprocessing                          102    102     0%   44-277
multiprocessing.process                  157    157     0%   35-323
multiprocessing.util                     173    173     0%   35-350
new                                        9      9     0%   6-17
nose                                       7      7     0%   1-10
nose.case                                204    204     0%   5-397
nose.config                              350    350     0%   1-661
nose.core                                170    170     0%   3-341
nose.exc                                   2      2     0%   8-9
nose.failure                              25     25     0%   1-42
nose.importer                            101    101     0%   7-167
nose.inspector                           122    122     0%   4-207
nose.loader                              347    347     0%   9-615
nose.plugins                               6      6     0%   184-190
nose.plugins.allmodules                   11     11     0%   20-45
nose.plugins.attrib                      103    103     0%   103-286
nose.plugins.base                        195    195     0%   1-724
nose.plugins.builtin                      12     12     0%   4-33
nose.plugins.capture                      56     56     0%   12-114
nose.plugins.collect                      42     42     0%   13-93
nose.plugins.cover                       145    145     0%   13-253
nose.plugins.debug                        31     31     0%   7-67
nose.plugins.deprecated                   16     16     0%   9-45
nose.plugins.doctests                    219    219     0%   50-454
nose.plugins.errorclass                   65     65     0%   95-210
nose.plugins.failuredetail                20     20     0%   9-48
nose.plugins.isolate                      34     34     0%   38-102
nose.plugins.logcapture                  126    126     0%   18-243
nose.plugins.manager                     265    265     0%   51-460
nose.plugins.multiprocess                490    490     0%   96-835
nose.plugins.plugintest                  155    155     0%   98-416
nose.plugins.prof                         93     93     0%   13-154
nose.plugins.skip                         22     22     0%   9-62
nose.plugins.xunit                       161    161     0%   39-321
nose.proxy                               108    108     0%   18-187
nose.pyversion                           131    131     0%   4-212
nose.result                              131    131     0%   11-198
nose.selector                            134    134     0%   9-250
nose.suite                               395    395     0%   10-605
nose.tools                                 5      5     0%   10-15
nose.tools.nontrivial                     80     80     0%   3-151
nose.tools.trivial                        24     24     0%   8-54
nose.util                                347    347     0%   3-661
nosexcover                                 1      1     0%   1
nosexcover.nosexcover                     25     25     0%   5-56
ntpath                                   278    278     0%   2-512
numbers                                  132    132     0%   4-391
opcode                                   156    156     0%   2-192
optparse                                 770    770     0%   24-1704
os                                       376    376     0%   26-740
pdb                                      856    856     0%   3-1338
pickle                                   859    859     0%   27-1391
pkg_resources                           1486   1486     0%   17-2762
pkgutil                                  321    321     0%   6-591
platform                                 742    742     0%   3-1678
posixpath                                235    235     0%   13-446
posixpath                                235    235     0%   13-446
pprint                                   229    229     0%   11-346
profile                                  302    302     0%   9-610
pstats                                   494    494     0%   23-703
pty                                      119    119     0%   9-180
qgis                                      31     31     0%   3-66
random                                   388    388     0%   42-905
raven                                      7      0   100%   
raven.base                               256    193    25%   37-44, 52-54, 57-65, 68-70, 73-75, 78, 83-85, 88, 91-93, 141-201, 205, 208-209, 218, 221, 231-332, 335, 392-400, 403-406, 410-416, 423, 426-447, 453-461, 469-491, 497, 503, 506-508, 512-514, 523, 539, 547, 557, 563
raven.conf                                25     20    20%   35-49, 74-86
raven.conf.defaults                       20      0   100%   
raven.transport                            3      0   100%   
raven.transport.base                     129     85    34%   16-17, 24-25, 29-30, 48-49, 56, 64, 72-74, 77-98, 101-121, 129-133, 139-145, 148-170, 178-185, 193-194, 197, 205-212, 215-216, 225-231, 234-243
raven.transport.exceptions                 2      0   100%   
raven.transport.registry                  35     12    66%   20, 30, 37, 40-49, 56-57
raven.transport.threaded                  65     44    32%   24-30, 33-42, 48-56, 62-69, 72, 75-81, 89-92, 95-97, 100, 103
raven.utils                               81     68    16%   14-15, 27-40, 48-70, 74-107, 111, 115-125
raven.utils.encoding                      47     41    13%   19-53, 57-60, 64-73, 77-80, 84-94
raven.utils.json                          18     10    44%   16-22, 26, 30, 34
raven.utils.serializer                     2      0   100%   
raven.utils.serializer.base               77     24    69%   18-21, 28, 35, 41, 48, 55-61, 68, 75, 82, 89, 96, 99, 106, 113, 120, 127
raven.utils.serializer.manager            47     29    38%   23-24, 34-38, 45-76, 84-85
raven.utils.stacks                       121    109    10%   23-78, 86-100, 108-111, 119-126, 135-141, 150-157, 169-239
raven.utils.urlparse                       8      0   100%   
re                                       134    134     0%   17-327
repr                                     103    103     0%   3-132
rfc822                                   574    574     0%   74-1011
shutil                                   315    315     0%   7-555
simplejson                                79     57    28%   119-121, 125-129, 236-262, 346-357, 427, 484-505, 509-532, 547
simplejson.compat                         36     19    47%   7, 22-41
simplejson.decoder                       219    170    22%   13, 60-133, 145-234, 237-270, 368-374, 387-389
simplejson.encoder                       340    287    16%   11, 45-53, 60-82, 211, 214, 216, 218, 239, 250-268, 280-342, 357-361, 364-369, 390-628
simplejson.scanner                        87     57    34%   7, 33-42, 46-51, 55-62, 80-115, 118-121
site                                     335    335     0%   67-615
sitecustomize                              5      5     0%   2-7
socket                                   354    354     0%   4-573
sre_compile                              389    389     0%   11-518
sre_constants                            127    127     0%   12-263
sre_parse                                622    622     0%   11-802
ssl                                      268    268     0%   4-579
stat                                      66     66     0%   8-96
stream_utilities                         587      6    99%   168, 540, 1200-1204
string                                   223    223     0%   23-642
struct                                     3      3     0%   1-3
subprocess                               691    691     0%   10-1621
symbol                                    98     98     0%   3-114
sysconfig                                351    351     0%   4-698
tempfile                                 321    321     0%   20-622
textwrap                                 131    131     0%   8-425
third_party                                0      0   100%   
third_party.raven                          7      0   100%   
third_party.raven.conf                    25     20    20%   35-49, 74-86
third_party.raven.contrib                  0      0   100%   
third_party.raven.contrib.transports       0      0   100%   
third_party.raven.contrib.zerorpc         16     10    38%   33-34, 44-55
third_party.raven.handlers                 0      0   100%   
third_party.raven.scripts                  0      0   100%   
third_party.raven.transport                3      0   100%   
third_party.raven.utils                   81     68    16%   14-15, 27-40, 48-70, 74-107, 111, 115-125
third_party.raven.utils.serializer         2      0   100%   
third_party.simplejson                    79     57    28%   119-121, 125-129, 236-262, 346-357, 427, 484-505, 509-532, 547
third_party.simplejson.compat             36     19    47%   7, 22-41
third_party.simplejson.decoder           219    170    22%   13, 60-133, 145-234, 237-270, 368-374, 387-389
third_party.simplejson.encoder           340    287    16%   11, 45-53, 60-82, 211, 214, 216, 218, 239, 250-268, 280-342, 357-361, 364-369, 390-628
third_party.simplejson.scanner            87     57    34%   7, 33-42, 46-51, 55-62, 80-115, 118-121
threading                                596    596     0%   3-1331
token                                    117    117     0%   11-140
tokenize                                 248    248     0%   25-432
traceback                                178    178     0%   3-320
tty                                       25     25     0%   5-36
twisted                                   31     31     0%   7-62
twisted._version                           3      3     0%   6-11
twisted.cred                               1      1     0%   6
twisted.cred._digest                      49     49     0%   5-129
twisted.cred.credentials                 157    157     0%   5-492
twisted.cred.error                         6      6     0%   5-34
twisted.internet                           1      1     0%   4
twisted.internet._baseprocess             35     35     0%   5-62
twisted.internet._newtls                  73     73     0%   5-271
twisted.internet._posixstdio             109    109     0%   3-175
twisted.internet._signals                 13     13     0%   5-68
twisted.internet._sslverify              309    309     0%   6-809
twisted.internet.abstract                186    186     0%   5-535
twisted.internet.address                  72     72     0%   4-181
twisted.internet.base                    563    563     0%   5-1213
twisted.internet.default                  18     18     0%   5-56
twisted.internet.defer                   495    495     0%   5-1607
twisted.internet.endpoints               539    539     0%   5-1771
twisted.internet.epollreactor            152    152     0%   4-410
twisted.internet.error                   159    159     0%   4-462
twisted.internet.fdesc                    44     44     0%   6-118
twisted.internet.interfaces              215    215     0%   4-2480
twisted.internet.main                     13     13     0%   6-37
twisted.internet.posixbase               284    284     0%   5-640
twisted.internet.process                 464    464     0%   5-1072
twisted.internet.protocol                288    288     0%   5-845
twisted.internet.ssl                      76     76     0%   6-198
twisted.internet.stdio                     8      8     0%   5-35
twisted.internet.task                    306    306     0%   5-899
twisted.internet.tcp                     474    474     0%   5-1181
twisted.internet.threads                  36     36     0%   4-126
twisted.internet.udp                     207    207     0%   5-433
twisted.persisted                          1      1     0%   4
twisted.persisted.styles                 141    141     0%   7-262
twisted.plugin                           117    117     0%   6-255
twisted.protocols                          1      1     0%   5
twisted.protocols.basic                  301    301     0%   6-952
twisted.protocols.policies               334    334     0%   5-727
twisted.protocols.tls                    223    223     0%   5-622
twisted.python                             1      1     0%   6
twisted.python._reflectpy3               122    122     0%   5-387
twisted.python.compat                    197    197     0%   7-408
twisted.python.components                140    140     0%   6-439
twisted.python.constants                 113    113     0%   5-428
twisted.python.context                    38     38     0%   5-133
twisted.python.deprecate                 134    134     0%   5-539
twisted.python.failure                   266    266     0%   8-654
twisted.python.filepath                  424    424     0%   5-1547
twisted.python.lockfile                  118    118     0%   6-213
twisted.python.log                       256    256     0%   5-611
twisted.python.modules                   261    261     0%   5-775
twisted.python.randbytes                  53     53     0%   5-150
twisted.python.reflect                   151    151     0%   5-286
twisted.python.runtime                    59     59     0%   5-178
twisted.python.systemd                    28     28     0%   5-87
twisted.python.threadable                 63     63     0%   5-139
twisted.python.util                      535    535     0%   5-1089
twisted.python.versions                   99     99     0%   5-258
twisted.python.win32                      68     68     0%   5-169
twisted.python.zippath                   105    105     0%   5-268
twisted.web                                3      3     0%   5-14
twisted.web._newclient                   505    505     0%   5-1604
twisted.web._responses                    46     46     0%   5-59
twisted.web._version                       3      3     0%   6-11
twisted.web.client                       724    724     0%   5-1807
twisted.web.error                        101    101     0%   5-380
twisted.web.http                         831    831     0%   5-1931
twisted.web.http_headers                  65     65     0%   5-261
twisted.web.iweb                          76     76     0%   5-703
types                                     55     55     0%   5-84
unittest                                  11     11     0%   47-69
unittest.case                            578    578     0%   3-1076
unittest.loader                          197    197     0%   3-316
unittest.main                            133    133     0%   3-236
unittest.result                          118    118     0%   3-191
unittest.runner                          148    148     0%   3-196
unittest.signals                          47     47     0%   1-71
unittest.suite                           208    208     0%   3-303
unittest.util                            120    120     0%   2-156
urllib                                  1066   1066     0%   25-1621
urllib2                                  820    820     0%   92-1470
urlparse                                 241    241     0%   31-428
uuid                                     300    300     0%   47-575
warnings                                 248    248     0%   6-400
weakref                                  280    280     0%   12-443
xml                                       15     15     0%   20-41
xml.dom                                   79     79     0%   18-139
xml.dom.NodeFilter                        19     19     0%   4-27
xml.dom.domreg                            53     53     0%   5-99
xml.dom.minicompat                        45     45     0%   39-110
xml.dom.minidom                         1348   1348     0%   18-1941
xml.dom.xmlbuilder                       213    213     0%   3-386
xml.sax                                   52     52     0%   22-108
xml.sax._exceptions                       46     46     0%   2-125
xml.sax.handler                           45     45     0%   12-337
xml.sax.saxutils                         212    212     0%   6-350
xml.sax.xmlreader                        161    161     0%   4-381
zipfile                                  935    935     0%   4-1527
zope                                       6      6     0%   2-7
zope.interface                            30     30     0%   14-89
zope.interface._compat                     3      3     0%   14-17
zope.interface.adapter                   443    443     0%   14-704
zope.interface.advice                     66     66     0%   14-205
zope.interface.declarations              308    308     0%   12-848
zope.interface.exceptions                 21     21     0%   14-62
zope.interface.interface                 381    381     0%   14-712
zope.interface.interfaces                140    140     0%   14-1283
zope.interface.ro                         25     25     0%   14-69
--------------------------------------------------------------------
TOTAL                                  60210  58943     2%   
----------------------------------------------------------------------
Ran 26 tests in 14.477s

FAILED (failures=2)

-----------
PEP8 issues
-----------
./stream_utilities.py:1077:11: E271 multiple spaces after keyword

-----------------
Pylint violations
-----------------
************* Module stream-feature-extractor
I: 25, 0: Locally disabling W0611 (locally-disabled)
C:  1, 0: Invalid module name "stream-feature-extractor" (invalid-name)
W: 27, 0: Relative import 'custom_logging', should be 'stream-feature-extractor.custom_logging' (relative-import)
C: 30, 0: Invalid constant name "sentry_url" (invalid-name)
C: 37, 0: Invalid function name "classFactory" (invalid-name)
W: 39, 4: Relative import 'stream_feature_extractor', should be 'stream-feature-extractor.stream_feature_extractor' (relative-import)

```
# Expected outcome

All tests should pass

**Comment by ismailsunni** [comment]
Hi @timlinux 

I'm not sure, but all tests pass in mine. I tried on console and pycharm,

```
----------------------
Regression Test Suite
----------------------
#1 Test that the plugin __init__ will validate on plugins.qgis.org. ... ok
#2 Test that QGIS properly parses a wkt string. ... Loading a style file that was saved with an older version of qgis (saved in 1.9.0-Master, loaded in 2.2.0-Valmiera). Problems may occur.
ok
#3 QGIS environment has the expected providers ... ok
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
#31 test for add_associated_nodes ... ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ok
#32 Test for check_associated_attributes ... ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ERROR 1: Invalid index : 3
ERROR 1: Invalid index : 4
ERROR 1: Invalid index : 5
ERROR 1: Invalid index : 6
ok
#33 Test for checking the data test. ... ok
#34 Test for creating nodes layer. ... ok
#35 Test for extracting nodes. ... ok
#36 Test for extracting nodes using dgn dataset. ... /usr/lib/python2.7/unittest/case.py:338: RuntimeWarning: TestResult has no addExpectedFailure method, reporting as passes
  RuntimeWarning)
ok
#37 Test for get_nearby_nodes function. ... ok
#38 Test for identify_branch method. ... ok
#39 Test for identify_confluence method. ... ok
#40 Test for identify_features on the dgn test dataset. ... ok
#41 Test for identify_features on the sungai baru dataset. ... ok
#42 Test for identify_pseudo_node method. ... ok
#43 Test for identify_segment_center. ... ok
#44 Test for identify_self_intersections. ... ok
#45 Test for identify_sink method. ... ok
#46 Test for identify_watershed method. ... ok
#47 Test for identify_well method. ... ok
#48 Test identify_intersections. ... ok
#49 test add_layer_attribute. ... ok
#50 test for list to str. ... ok
#51 test for get_random_string function. ... ok
#52 test for str_to_list. ... ok
#4 Test that translations work. ... ok

```

```
--------------------------------------------------------------------
TOTAL                                  60292  53668    11%   
----------------------------------------------------------------------
Ran 26 tests in 19.041s

OK

-----------
PEP8 issues
-----------

-----------------
Pylint violations
-----------------

```

**Comment by ismailsunni** [comment]
Closed by 35d8a04
